src/overview/components/DateRangeSelection.jsx

```diff
@@ -1,46 +1,126 @@
-import React, { PropTypes } from 'react'
+import React, { Component, PropTypes } from 'react'
 import moment from 'moment'
+import chrono from 'chrono-node'
 import DatePicker from 'react-datepicker'
 import 'react-datepicker/dist/react-datepicker-cssmodules.css'
 
 import styles from './DateRangeSelection.css'
 
 
-const DateRangeSelection = ({
-    startDate, endDate,
-    onStartDateChange, onEndDateChange,
-}) => (
-    <div className={styles.dateRangeSelection}>
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='after..'
-            isClearable
-            selected={startDate && moment(startDate)}
-            selectsStart
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onStartDateChange(
-                date ? date.valueOf() : undefined,
-            )}
-        />
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='before..'
-            isClearable
-            selected={endDate && moment(endDate)}
-            selectsEnd
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onEndDateChange(
-                date ? date.endOf('day').valueOf() : undefined,
-            )}
-        />
-    </div>
-)
+class DateRangeSelection extends Component {
+    constructor() {
+        super()
+
+        this.state = {
+            startDateText: null,
+            endDateText: null,
+        }
+
+        this.handleKeydown = this.handleKeydown.bind(this)
+        this.submitDateChange = this.submitDateChange.bind(this)
+    }
+
+    componentDidMount() {
+        this.refs.startDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'startDate'))
+        this.refs.endDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'endDate'))
+    }
+
+    handleKeydown(event, type) {
+        if (event.key === 'Enter') {
+            event.stopImmediatePropagation()
+            this.submitDateChange(type)
+        }
+    }
+
+    submitDateChange(type) {
+        if (type === 'startDate') {
+            console.log('hrere')
+            const date = moment(this.state.startDateText, 'DD-MM-YYYY')
```

Comment:
"chrono-node" when trying to get IT formats does it like this "MM-DD-YYYY". Everywhere else in this component we use "DD-MM-YYYY" and I thought it was very confusing to use two different formats. Also moment is not going away since it's used everywhere in this component and needed for "react-datepicker" so might as well just use especially if it makes code simpler.

I think it would be very confusing to someone why there are months and dates inverted at least it was to me until I found out.. Also means we would had to set the state in a different format.